src/wp-includes/fonts/class-wp-font-face.php

```diff
@@ -0,0 +1,400 @@
+<?php
+/**
+ * WP_Font_Face class.
+ *
+ * @package    WordPress
+ * @subpackage Fonts
+ * @since      6.4.0
+ */
+
+/**
+ * Font Face generates and prints `@font-face` styles for given fonts.
+ *
+ * @since 6.4.0
+ */
+class WP_Font_Face {
+
+	/**
+	 * The font-face property defaults.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var string[]
+	 */
+	private $font_face_property_defaults = array(
+		'font-family'  => '',
+		'font-style'   => 'normal',
+		'font-weight'  => '400',
+		'font-display' => 'fallback',
+	);
+
+	/**
+	 * Valid font-face property names.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var string[]
+	 */
+	private $valid_font_face_properties = array(
+		'ascent-override',
+		'descent-override',
+		'font-display',
+		'font-family',
+		'font-stretch',
+		'font-style',
+		'font-weight',
+		'font-variant',
+		'font-feature-settings',
+		'font-variation-settings',
+		'line-gap-override',
+		'size-adjust',
+		'src',
+		'unicode-range',
+	);
+
+	/**
+	 * Valid font-display values.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var string[]
+	 */
+	private $valid_font_display = array( 'auto', 'block', 'fallback', 'swap', 'optional' );
+
+	/**
+	 * Array of font-face style tag's attribute(s)
+	 * where the key is the attribute name and the
+	 * value is its value.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var string[]
+	 */
+	private $style_tag_attrs = array();
+
+	/**
+	 * Creates and initializes an instance of WP_Font_Face.
+	 *
+	 * @since 6.4.0
+	 */
+	public function __construct() {
+		if (
+			function_exists( 'is_admin' ) && ! is_admin()
+			&&
+			function_exists( 'current_theme_supports' ) && ! current_theme_supports( 'html5', 'style' )
+		) {
+			$this->style_tag_attrs = array( 'type' => 'text/css' );
+		}
+	}
+
+	/**
+	 * Generates and prints the `@font-face` styles for the given fonts.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array[][] $fonts Optional. The font-families and their font variations.
+	 *                         See {@see wp_print_font_faces()} for the supported fields.
+	 *                         Default empty array.
+	 * }
```

Comment:
Nitpick: this curly brace doesn't seem to be necessary here.
```suggestion
	 *
```

Reply:
Thank you @anton-vlasenko! Done in https://github.com/WordPress/wordpress-develop/pull/5051/commits/4a1d4c2b55ee54cb8563ad06ed9237ce5f73ab6a ✅

Reply:
Thank you!